File: samples/scoping-samples/simple-stack-example-scoping-java/src/main/java/com/zhuinden/simplestackexamplescoping/features/words/NewWordFragment.java
package com.zhuinden.simplestackexamplescoping.features.words;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import com.zhuinden.simplestack.Backstack;
import com.zhuinden.simplestack.navigator.Navigator;
import com.zhuinden.simplestackexamplescoping.R;
import com.zhuinden.simplestackextensions.fragments.KeyedFragment;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class NewWordFragment
        extends KeyedFragment {
    public NewWordFragment() {
        super(R.layout.new_word_fragment);
    }

    public interface ActionHandler {
        void onAddWordClicked(String word);
    }

    private ActionHandler actionHandler;

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        Backstack backstack = Navigator.getBackstack(requireContext());
        actionHandler = backstack.lookupService(ActionHandler.class.getName());

        view.findViewById(R.id.buttonAddNewWord).setOnClickListener(v -> {
            String word = view.<TextView>findViewById(R.id.textInputNewWord).getText().toString().trim();
            actionHandler.onAddWordClicked(word);
        });
    }
}

File: samples/scoping-samples/simple-stack-example-scoping/src/main/java/com/zhuinden/simplestackexamplescoping/features/words/NewWordFragment.kt
package com.zhuinden.simplestackexamplescoping.features.words

import android.os.Bundle
import android.view.View
import com.zhuinden.simplestackexamplescoping.R
import com.zhuinden.simplestackexamplescoping.databinding.NewWordFragmentBinding
import com.zhuinden.simplestackexamplescoping.utils.onClick
import com.zhuinden.simplestackexamplescoping.utils.viewBinding
import com.zhuinden.simplestackextensions.fragments.KeyedFragment
import com.zhuinden.simplestackextensions.fragmentsktx.lookup


/**
 * Created by Zhuinden on 2018.09.17.
 */

class NewWordFragment : KeyedFragment(R.layout.new_word_fragment) {
    private val binding by viewBinding(NewWordFragmentBinding::bind)

    interface ActionHandler {
        fun onAddWordClicked(word: String)
    }

    private val actionHandler by lazy { lookup<ActionHandler>() }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.buttonAddNewWord.onClick {
            val word = binding.textInputNewWord.text.toString().trim()
            actionHandler.onAddWordClicked(word)
        }
    }
}
